File: go/rna-transcription/rna_transcription.go
package strand

// DNAtoRNA provides the mapping of DNA nucleotides to RNA nucleotides.
var DNAtoRNA = map[string]string{
	"G": "C",
	"C": "G",
	"T": "A",
	"A": "U",
}

// ToRNA converts a DNA strand to an RNA strand.
func ToRNA(dna string) string {
	var result string

	for _, nucleotide := range dna {
		result += DNAtoRNA[string(nucleotide)]
	}

	return result
}

File: elixir/rna-transcription/lib/rna_transcription.ex
defmodule RnaTranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RnaTranscription.to_rna('ACTG')
  'UGAC'
  """

  @spec to_rna(charlist()) :: charlist()
  def to_rna(dna) do
    dna |> Enum.map(&dna_to_rna/1)
  end

  defp dna_to_rna(?G), do: ?C
  defp dna_to_rna(?C), do: ?G
  defp dna_to_rna(?T), do: ?A
  defp dna_to_rna(?A), do: ?U
end
